jep/0000/README.adoc

```diff
@@ -0,0 +1,187 @@
+
+This specification describes the problem, and the overall proposed solution,
+as it relates to the Jenkins project and existing structures.
+
+The details of the exact behavior and implementation of most areas will be addressed in one or more separate JEPs.
+
+=== Target Audience
+
+Jenkins Operator is targeted at existing and new Jenkins users (system administrators, DevOps engineers or regular programmers)
+who are not necessarily interested in becoming Jenkins experts, but rather wish to use Jenkins as a means to an end.
+That end being: *building, testing, and delivering their projects*, with as *little overhead* as possible.
+Potential users wish to use a stable and reliable CI/CD tool.
+
+Jenkins Operator is going to be the best solution for everybody who is interested in running Jenkins
+on Kubernetes cluster.
+
+=== Goals
+
+==== Run Jenkins on top of Kubernetes
+
+Use link:https://coreos.com/blog/introducing-operator-framework[operator pattern] to manage complex Jenkins stateful
+application on top of Kubernetes.
+
+==== Jenkins configuration in one place
+
+Put all information needed to configure Jenkins in a single
+link:https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/[CRD] Kubernetes resource.
+
+==== Keep Jenkins instance configured
+
+Watch all changes made to Jenkins configuration
+link:https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/[CR] and apply it to the live
+instance.
+
+==== Make Jenkins immutable
+
+Jenkins master instance should be restorable to the desired (configured) state at any point in time.
+Any changes made to the Jenkins configuration that are *not* in
+link:https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/[CR]
+should be removed after Jenkins master restart to *avoid snowflake* problem.
+
+==== Scale Jenkins agents
+
+Spawn agents on demand to scale Jenkins workloads.
+
+==== Make default configuration secure
+
+Provide secure configuration as the default option.
+
+==== Make backup of Jenkins jobs history
+
+Backup Jenkins jobs history using tools provided by Kubernetes itself and optionally a cloud provider.
+
+==== Make errors more visible for end users
+
+Validate and visualise problems in configuration provided by the user.
+
+==== Reduce Jenkins master unavailability
+
+== Motivation
+
+Jenkins Operator contains complex knowledge about operating Jenkins, in particular running it on Kubernetes.
+Given that it shouldn't be a part Jenkins implementation.
+
+== Reasoning
+
+This document does not propose a specific design or implementation, as other JEPs might. This is intentional.
+This document specifies the concept of "Jenkins Operator" and the mission and priorities of the project.
+Later related JEPs will address design and implementation. Limiting the scope of this JEP to high-level goals will let
+us build a solid consensus for that direction separate from the design of individual features.
```

Comment:
@bitwiseman what was the outcome of our discussion of “meta-JEPs”? I thought we had agreed to discourage them and instead use SIGs for that purpose, limiting actual JEPs to concrete proposals. But not sure if these guidelines were ever solidified and written down.

Reply:
@jglick please look at new commit